examples/chatbot.py

```diff
@@ -0,0 +1,191 @@
+import os
+import openai
+import nest_asyncio
+import certifi
+import nltk
+from pathlib import Path
+from llama_index.readers.file import UnstructuredReader
+from llama_index.core import VectorStoreIndex, StorageContext, Settings, load_index_from_storage
+from llama_index.core.tools import QueryEngineTool, ToolMetadata
+from llama_index.llms.openai import OpenAI
+from llama_index.core.query_engine import SubQuestionQueryEngine
+from llama_index.agent.openai import OpenAIAgent
+from aimon.client import Client
+from llama_index.embeddings.openai import OpenAIEmbedding
+from dotenv import load_dotenv
+import requests
+import json
+from aimon import Config
+import logging
+import time
+
+load_dotenv()
+
+email = os.getenv('EMAIL')
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logging.basicConfig(level=logging.DEBUG)
+
+nest_asyncio.apply()
+os.environ['SSL_CERT_FILE'] = certifi.where()
+
+# Download UBER dataset
+os.system("mkdir -p data")
+os.system('wget "https://www.dropbox.com/s/948jr9cfs7fgj99/UBER.zip?dl=1" -O data/UBER.zip')
+os.system("unzip -o data/UBER.zip -d data")
+nltk.download('averaged_perceptron_tagger')
+
+# Load UBER documents
+years = [2022, 2021, 2020, 2019]
+loader = UnstructuredReader()
+doc_set = {}
+all_docs = []
+index_set = {}
+
+for year in years:
+    year_docs = loader.load_data(file=Path(f"./data/UBER/UBER_{year}.html"), split_documents=False)
+    # Insert year metadata into each document
+    for d in year_docs:
+        d.metadata = {"year": year}
+    doc_set[year] = year_docs
+    all_docs.extend(year_docs)
+
+Settings.chunk_size = 512
+Settings.chunk_overlap = 64
+Settings.llm = OpenAI(model="gpt-3.5-turbo")
+Settings.embed_model = OpenAIEmbedding(model="text-embedding-3-small")
+
+# Initialize simple vector indices
+index_set = {}
+for year in years:
+    storage_context = StorageContext.from_defaults()
+    cur_index = VectorStoreIndex.from_documents(
+        doc_set[year],
+        storage_context=storage_context,
+    )
+    index_set[year] = cur_index
+    storage_context.persist(persist_dir=f"./storage/{year}")
+
+individual_query_engine_tools = [
```

Comment:
line 69 to line 92 can be refactored in to a `construct_agent()` function

Reply:
done